test/e2e/pkg/testnet.go

```diff
@@ -466,7 +471,17 @@ func newKeyGenerator(seed int64) *keyGenerator {
 	}
 }
 
-func (g *keyGenerator) Generate() crypto.PrivKey {
+func (g *keyGenerator) Generate(keyType string) crypto.PrivKey {
+	if keyType == "secp256k1" {
+		var seed []byte
+		_, err := io.ReadFull(g.random, seed)
+		if err != nil {
+			panic(err) // this shouldn't happen
+		}
+
+		return secp256k1.GenPrivKeySecp256k1(seed)
+
+	}
```

Comment:
This should be a switch. Also, seed generation code is the same across key types.